pkg/apis/garden/validation/validation_shoot.go

```diff
@@ -957,15 +957,8 @@ func validateDNSUpdate(new, old *garden.DNS, seedGotAssigned bool, fldPath *fiel
 		// allow to finalize DNS configuration during seed assignment. this is required because
 		// some decisions about the DNS setup can only be taken once the target seed is clarified.
 		if !seedGotAssigned {
-			providersNew := 0
-			if new != nil {
-				providersNew = len(new.Providers)
-			}
-
-			providersOld := 0
-			if old != nil {
-				providersOld = len(old.Providers)
-			}
+			providersNew := len(new.Providers)
```

Comment:
What if `new` is `nil`?

Reply:
It is checked just a few lines above https://github.com/gardener/gardener/pull/1778/files#diff-64343c81d87f4d241b69590abc33db52R952.

Reply:
Thanks!